### ISSUE TITLE
Evaluate disabling some JAI ImageIO plugins

### ISSUE THREAD

**Issue by lfcnassif** [sepinf-inc/IPED#1134]
I've just fixed some "cannot seek" exceptions being thrown in #1116 ([3474322](https://github.com/sepinf-inc/IPED/pull/1116/commits/3474322723c3001ab9f86638320637afb3444c9c)) caused by JAI ImageIO TIFFImageReader being used in that branch. On current master, the internal Sun impl is used and the exceptions are not thrown while processing, but when opening UI, the JAI impl is used and causes exceptions to be thrown, making behavior inconsistent (this took me some time to debug...).

Java 9 added support for TIFF. JAI plugin has some intersections with formats already supported by the JVM, I think we should evaluate which implementations are better and disable the others, so the dynamic service loader behavior won't change results...

**Comment by lfcnassif** [comment]
@tc-wleite, out of curiosity, have you done some related study in the past? (I have a far remind about that...)

**Comment by tc-wleite** [comment]
> @tc-wleite, out of curiosity, have you done some related study in the past? (I have a far remind about that...)

Well, I don't remember :-)
I did some comparisons, but I think they didn't involve JAI. And definitely didn't test with Java >= 9.

That being said, I think the intersection you mentioned is a valid concern. Removing JAI altogether would be great, but probably won't cover all the formats, right?

**Comment by lfcnassif** [comment]
> but probably won't cover all the formats, right?

Yes, they support pcx, pnm & raw, not supported by the JVM, and even other formats could have different codec support...

I've also seen NPEs being thrown when parsing BMP files for a while. I'll do some testing to check which plugin is the bad guy, and if the other process the triggering files fine. But switching would require a larger test...

**Comment by lfcnassif** [comment]
All 1147 NPEs (from BMPs in the 33M dataset) comes from JAI. Removing them from classpath, causes different IOExceptions to be thrown by the internal JVM BMP image reader, from those BMPs. All of them are corrupted. Although NPEs commonly are symptoms of bugs, this BMP issue doesn't seem critical to me.

PS: Tika 2 causes JAI impls to take priority, maybe they changed something in service loading...

**Comment by lfcnassif** [comment]
Just hit some NPEs running regression tests while processing (corrupted) BMPs again, caused by JAI. Today we have 3 BMP decoder providers: JAI, the JVM default and TwelveMonkeys. My concern is that ImageIO service loading seems to be not deterministic, linking with TSK-4.11.1 those exceptions weren't thrown, but linking with TSK-4.12.0 they were...

**Comment by lfcnassif** [comment]
I could confirm the exceptions stopped to be thrown excluding JAI from classpath. As all those BMPs are corrupted and doesn't render, I think this is not critical, but we should have a deterministic behavior...
